### PR TITLE
feat: add qdldl

### PR DESCRIPTION
**Pull request by wep21**
- add [qdldl](https://github.com/osqp/qdldl/releases/tag/v0.1.7)

### CODE REVIEW

**Comment by wep21** [comment]
Need skip-url-stability-check label.

**Review by fmeum — APPROVED** [review]

File: modules/qdldl/0.1.7/patches/add_cmake_configure_file.patch [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
++        env = {},
++        **kwargs
++    )
+diff --git bazel/cmake_configure_file.py bazel/cmake_configure_file.py
+new file mode 100644
+index 0000000..7097821
+--- /dev/null
++++ bazel/cmake_configure_file.py
+@@ -0,0 +1,226 @@
++"""A re-implementation of CMake's configure_file substitution semantics.  This
```
Comment (fmeum) [inline modules/qdldl/0.1.7/patches/add_cmake_configure_file.patch:111]: Are you thinking of contributing this rule to a repo such as `rules_foreign_cc`? That could simplify the setup for new modules.

File: modules/qdldl/0.1.7/patches/add_build_file.patch [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
++    copts = [
++        "-fvisibility=hidden",
++    ] + select({
++        "@platforms//os:windows": [],
++        "//conditions:default": ["-Werror=incompatible-pointer-types"],
++    }),
++    includes = [
++        "include",
++    ],
++    linkstatic = 1,
```
Comment (fmeum) [inline modules/qdldl/0.1.7/patches/add_build_file.patch:70]: Is this necessary? It does restrict users.

Comment (wep21) [reply modules/qdldl/0.1.7/patches/add_build_file.patch]: remove at https://github.com/bazelbuild/bazel-central-registry/pull/1421/commits/4d48b5640e78bce45a082232f37ea787b692f8b0

**Pull request merged**